src/Microsoft.TemplateEngine.Edge/FileSystem/InMemoryFileSystem.cs

```diff
@@ -0,0 +1,917 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Text.RegularExpressions;
+using Microsoft.TemplateEngine.Abstractions.PhysicalFileSystem;
+
+namespace Microsoft.TemplateEngine.Edge.FileSystem
+{
+    public class InMemoryFileSystem : IPhysicalFileSystem, IFileLastWriteTimeSource
+    {
+        private readonly FileSystemDirectory _root;
+        private readonly IPhysicalFileSystem _basis;
+
+        public InMemoryFileSystem(string root, IPhysicalFileSystem basis)
+        {
+            _basis = basis;
+            _root = new FileSystemDirectory(Path.GetFileName(root.TrimEnd('/', '\\')), root);
+            IsPathInCone(root, out root);
+            _root = new FileSystemDirectory(Path.GetFileName(root.TrimEnd('/', '\\')), root);
```

Comment:
Shouldn't this be done only if `IsPathInCone` actually changed root? Seems wasteful otherwise.